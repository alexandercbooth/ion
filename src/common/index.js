// @flow
export * from "./classes";
export * from "./elevation";
export * from "./intent";
