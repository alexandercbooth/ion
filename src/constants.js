// @flow
export const Elevation = {
  LOWEST: 1,
  MEDIUM: 2,
  HIGHEST: 3
};
