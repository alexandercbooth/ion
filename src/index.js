// @flow
export * from "./common";
export { Card } from "./components/card";
export { TextArea } from "./components/textArea";
