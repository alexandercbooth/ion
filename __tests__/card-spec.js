// @flow
import * as React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Elevation } from "../src/constants";
import { Card } from "../src";

describe("Card", () => {
  it("renders a div with elevation prop", () => {
    const component = shallow(
      <div>
        <Card elevation={Elevation.LOWEST}>
          <code>[1]</code>Test
        </Card>
        <Card elevation={Elevation.MEDIUM}>
          <code>[2]</code>The
        </Card>
        <Card elevation={Elevation.HIGHEST}>
          <code>[3]</code>Cards
        </Card>
      </div>
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
