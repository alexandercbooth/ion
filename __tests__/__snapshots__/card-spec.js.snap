// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders a div with elevation prop 1`] = `
<div>
  <Card
    elevation={1}
  >
    <code>
      [1]
    </code>
    Test
  </Card>
  <Card
    elevation={2}
  >
    <code>
      [2]
    </code>
    The
  </Card>
  <Card
    elevation={3}
  >
    <code>
      [3]
    </code>
    Cards
  </Card>
</div>
`;
